src/index.ts

```diff
@@ -1,14 +1,26 @@
-export * from './debugging'
-import { Config } from '@algorandfoundation/algokit-utils'
-import { DebugParams } from '@algorandfoundation/algokit-utils/types/debugging'
-import { persistSourceMaps, simulateAndPersistResponse } from './debugging'
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { AVMTracesEventData, Config, TealSourcesDebugEventData } from '@algorandfoundation/algokit-utils'
+import { EventType } from '@algorandfoundation/algokit-utils/types/async-event-emitter'
+import { writeAVMDebugTrace, writeTealDebugSourceMaps } from './debugging'
 
-// Automatically register debug handlers upon import
-Config.registerDebugHandler(async (params: DebugParams) => {
-  if (params.message === 'persistSourceMaps') {
-    await persistSourceMaps(params.data)
-  } else if (params.message === 'simulateAndPersistResponse') {
-    await simulateAndPersistResponse(params.data)
-  }
-  // Add more handlers as needed
-})
+/**
+ * Registers event handlers for debugging events in Algorand development.
+ *
+ * This function sets up listeners for the following events:
```

Comment:
```suggestion
 * This function sets up handlers for the following events:
```
To match `registerDebugEventHandlers`